app/questionnaire/questionnaire_schema.py

```diff
@@ -215,19 +215,36 @@ def get_post_submission(self) -> ImmutableDict:
         schema: ImmutableDict = self.json.get("post_submission", ImmutableDict({}))
         return schema
 
+    def _check_list_exists(self, rules: Union[dict, Sequence], list_name: str) -> bool:
```

Comment:
I think we should rename it to something like, `_is_list_name_in_rule`?. `_check_list_exists` to me suggests tell me if list exists in the schema.